### PR TITLE
Remove flex_category from stages table (lessons)

### PR DESCRIPTION
**Pull request by dmcavoy**
Migration to remove flex_category column from the stages table (lessons model).

This is one of the last step of removing flex_category besides the Script Edit Gui and removing the old flex_category translations.

### CODE REVIEW

**Review by davidsbailey — APPROVED** [review]
LGTM! Is this the PR which will break script edit gui? If so, it's possible you'll have an easier time moving it over to lesson groups before merging this PR, since it could make it easier to get the new behavior to match the old behavior. It's up to you, though.

**Comment by uponthesun** [comment]
Do we know this is safe to deploy in production? Might be worth checking in with the other infra team members, unless we've done this kind of thing before.

**Comment by dmcavoy** [comment]
@wjordan and @sureshc I added you to review this PR from Winter's suggestion. Is it safe to remove this field on production?

**Comment by davidsbailey** [comment]
> @wjordan and @sureshc I added you to review this PR from Winter's suggestion. Is it safe to remove this field on production?

for context, we believe we've removed any code used in production which depends on this column.

**Comment by sureshc** [comment]
I see several past migrations that drop a column from a table, so I think this is safe.  There appear to be about 58 rows in the production table with a value in this column.  Any need to backup this data?

**Comment by davidsbailey** [comment]
> I see several past migrations that drop a column from a table, so I think this is safe. There appear to be about 58 rows in the production table with a value in this column. Any need to backup this data?

not really, because it is already backed up in github, in the PR where we remove flex category from all the .script files.

**Review by sureshc — APPROVED** [review]

**Comment by sureshc** [comment]
Looks like there have been other migrations merged since this feature branch was created, so either re-base or merging the latest from `staging` is needed to resolve the merge conflict in `schema.rb`

File: dashboard/db/schema.rb [hunk 1]
```diff
@@ -1756,14 +1755,6 @@
...
   end
 
-  add_foreign_key "ap_school_codes", "schools"
-  add_foreign_key "census_inaccuracy_investigations", "census_overrides"
-  add_foreign_key "census_inaccuracy_investigations", "census_submissions"
-  add_foreign_key "census_inaccuracy_investigations", "users"
-  add_foreign_key "census_overrides", "schools"
-  add_foreign_key "census_submission_form_maps", "census_submissions"
-  add_foreign_key "census_summaries", "schools"
-  add_foreign_key "circuit_playground_discount_applications", "schools"
```
Comment (breville) [inline dashboard/db/schema.rb:1766]: These should not have been removed.  Restored in https://github.com/code-dot-org/code-dot-org/pull/34711.

**Pull request merged**